src/main/bootstrap/mod.rs

```diff
@@ -156,14 +152,22 @@ impl<UID: Uid> Bootstrap<UID> {
 
     fn handle_result(
         &mut self,
-        core: &mut Core,
+        core: &mut EventLoopCore,
         poll: &Poll,
         child: Token,
         res: Result<(TcpSock, PeerInfo, UID), (PeerInfo, Option<BootstrapDenyReason>)>,
     ) {
         let _ = self.children.remove(&child);
         match res {
-            Ok((socket, peer_addr, peer_id)) => {
+            Ok((socket, peer_info, peer_id)) => {
+                {
+                    let bootstrap_cache = core.user_data_mut();
+                    bootstrap_cache.put(&peer_info);
+                    if let Err(e) = bootstrap_cache.commit() {
+                        warn!("Failed to write bootstrap cache to disk: {}", e);
```

Comment:
Again, I'd say this is unexpected and should be inspected, so warn the user.

Reply:
true, `warn!` is for unexpected stuff but more for unexpected due to our logic (so if it errors out it's a bug in our code) as most errors (even the debug ones) are in a way, unexpected (hence errors). In addition to that `warn!` should also signify that this is potentially going to lead to UB and we would normally have terminated the app, but in the interest of not jeopardising every other connection immediately etc., we are just continuing - e.g, when we downcast `State` to say `ServiceDiscovery` and find out it failed - there's no way it can unless logic error or something so unexpected happened at the runtime that we can no longer guarantee determinism in behaviour of the running app any more.

Failure in bootstrap cache file handling is certainly something to notice, but at the same time not very ominous. Could be due to file permissions in the directory where crust is running etc.